### PR TITLE
feat: move page overview to sidebar

### PR DESCRIPTION
**Pull request by tilmx**
For easier page switches I moved the page overview to the left sidebar. Now you can easily switch between pages – like in Apple Keynote e.g.

## Content
* Move Page Overview from a separate View to a new Sidebar in PageDetail View
* Introduce Focus State for Elements and Pages (`focusedItem`)

## Todo
- [x] Move Page Overview
- [x] Set AppFocus on Element or Page Select
- [x] Select next page when deleting the very first page (Bug)
- [x] Show or Hide Page View with "< Pages" button
~~Make entire page tile editable~~ (Coming with PR #553)

## Bugs
- [x] First Open Project: DEL leads to weird unreversible deletion

### CODE REVIEW

File: src/components/add-page/demo.tsx [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import DemoContainer from '../demo-container';
+import * as React from 'react';
+import { AddPage } from '.';
+
+const AddPageDemo: React.StatelessComponent<void> = (): JSX.Element => (
```
Comment (marionebl) [inline src/components/add-page/demo.tsx]: Omit the `void` type parameter

File: src/components/add-page/demo.tsx [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+	<DemoContainer title="Floating Button">
```
Comment (marionebl) [inline src/components/add-page/demo.tsx]: Change to `Add Button`

File: src/components/add-page/pattern.json [hunk 3]
```diff
@@ -0,0 +1,8 @@
+{
+	"name": "add-page",
```
Comment (marionebl) [inline src/components/add-page/pattern.json]: add-page-button might be a more distinct name

File: src/container/element-list/element-container.tsx [hunk 4]
```diff
@@ -30,6 +30,7 @@ export class ElementContainer extends React.Component<ElementContainerProps> {
 			<Components.Element
 				draggable={true}
 				dragging={store.getDragging()}
+				editable={props.element.getRole() === ElementRole.Root ? false : true}
```
Comment (marionebl) [inline src/container/element-list/element-container.tsx]: `props.element.getRole() !== ElementRole.Root`

File: src/container/element-list/element-container.tsx [hunk 5]
```diff
@@ -38,7 +39,7 @@ export class ElementContainer extends React.Component<ElementContainerProps> {
 				onChange={AlvaUtil.noop}
 				placeholderHighlighted={props.element.getPlaceholderHighlighted()}
 				state={getState(props.element, store)}
-				title={props.element.getName()}
+				title={props.element.getRole() === ElementRole.Root ? 'Page' : props.element.getName()}
```
Comment (marionebl) [inline src/container/element-list/element-container.tsx]: Let's use the page name for the root element, should be implemented in `Element.prototype.getName`

File: src/container/page-list/page-list-container.tsx [hunk 6]
```diff
@@ -17,20 +19,27 @@ export const PageListContainer: React.StatelessComponent = MobxReact.inject('sto
...
-								page={page}
-							/>
-						))}
-					<PageAddButton />
-				</Layout>
-			</Space>
+			<Layout wrap={LayoutWrap.Wrap}>
+				{project
+					.getPages()
+					.map((page, i) => (
```
Comment (marionebl) [inline src/container/page-list/page-list-container.tsx]: Change to `map(page => {...})`

File: src/container/page-list/page-tile-container.tsx [hunk 7]
```diff
@@ -1,12 +1,13 @@
...
+import { PageTile } from '../../components';
 import * as MobxReact from 'mobx-react';
 import { Page } from '../../model';
 import * as React from 'react';
 import { ViewStore } from '../../store';
 import * as Types from '../../types';
 
 export interface PageTileContainerProps {
-	focused: boolean;
+	focus: boolean;
```
Comment (marionebl) [inline src/container/page-list/page-tile-container.tsx]: `focused`, `highlighted`

File: src/electron/renderer.tsx [hunk 8]
```diff
@@ -204,10 +200,17 @@ Sender.receive(message => {
 			break;
 		}
 		case ServerMessageType.Delete: {
-			if (app.getActiveView() === Types.AlvaView.Pages) {
+			if (
+				app.getActiveView() === Types.AlvaView.PageDetail &&
```
Comment (marionebl) [inline src/electron/renderer.tsx]: The "avctive view" concept should/could be removed entirely

Comment (tilmx) [reply src/electron/renderer.tsx]: At the moment this distinguishes between splashscreen and detail page – or would you handle that differently?

**Pull request merged**